Sources/VLCActionSheet.swift

```diff
@@ -0,0 +1,422 @@
+        addSubview(title)
+        NSLayoutConstraint.activate([
+            title.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 20),
+            title.centerYAnchor.constraint(equalTo: centerYAnchor),
+            title.centerXAnchor.constraint(equalTo: centerXAnchor),
+            title.topAnchor.constraint(equalTo: topAnchor)
+            ])
+    }
+}
+
+@objc protocol VLCActionSheetDataSource: class {
+    @objc func numberOfRows() -> Int
+    @objc func actionSheet(collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell
+}
+
+@objc protocol VLCActionSheetDelegate: class {
+    @objc optional func headerViewTitle() -> String?
+    @objc func itemAtIndexPath(_ indexPath: IndexPath) -> Any?
+    @objc func actionSheet(collectionView: UICollectionView, didSelectItem item: Any, At indexPath: IndexPath)
+}
+
+// MARK: VLCActionSheet
+class VLCActionSheet: UIViewController {
+
+    private let cellHeight: CGFloat = 50
+
+    @objc weak var dataSource: VLCActionSheetDataSource?
+    @objc weak var delegate: VLCActionSheetDelegate?
+
+    var action: ((_ item: Any) -> Void)?
+
+    lazy var backgroundView: UIView = {
+        let backgroundView = UIView()
+        backgroundView.isHidden = true
+        backgroundView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        backgroundView.backgroundColor = UIColor.black.withAlphaComponent(0.6)
+        backgroundView.isUserInteractionEnabled = true
+        backgroundView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.removeActionSheet)))
+        return backgroundView
+    }()
+
+    lazy var collectionViewLayout: VLCRendererCollectionViewLayout = {
+        let collectionViewLayout = VLCRendererCollectionViewLayout()
+        return collectionViewLayout
+    }()
+
+    lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: UIScreen.main.bounds, collectionViewLayout: collectionViewLayout)
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.backgroundColor = .white
+        collectionView.alwaysBounceVertical = true
+        collectionView.showsVerticalScrollIndicator = false
+        collectionView.register(VLCActionSheetCell.self, forCellWithReuseIdentifier: VLCActionSheetCell.identifier)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        return collectionView
+    }()
+
+    lazy var cancelButton: UIButton = {
+        let cancelButton = UIButton()
+        cancelButton.setTitle(NSLocalizedString("CANCEL", comment: ""), for: .normal)
+        cancelButton.setTitleColor(.white, for: .normal)
+        cancelButton.titleLabel?.font = UIFont.systemFont(ofSize: 16)
+        cancelButton.addTarget(self, action: #selector(self.removeActionSheet), for: .touchDown)
+        cancelButton.backgroundColor = UIColor(red:1.00, green:0.59, blue:0.13, alpha:1.0)
```

Comment:
please use a global VLC color for this or create a new one.

Reply:
Yes, hooking `VLCActionSheet` with `PresentationTheme` is in the todo list~